lib/membrane/aac/parser.ex

```diff
@@ -72,15 +71,23 @@ defmodule Membrane.AAC.Parser do
     end
   end
 
+  @impl true
   def handle_process(:input, buffer, ctx, state) when state.in_encapsulation == :none do
     # Since there is no ADTS header, there is nothing to parse on the input
     # Therefore, we only really add the timestamp
     timestamp = Helper.next_timestamp(state.timestamp, ctx.pads.input.caps)
     metadata = Map.put(buffer.metadata, :timestamp, timestamp)
 
+    buffer = %{buffer | metadata: metadata}
+
     buffer =
-      %{buffer | metadata: metadata}
-      |> then(&ensure_out_encapsulation(&1, ctx.pads.input.caps, state))
+      case state.out_encapsulation do
+        :ADTS ->
+          %Buffer{buffer | payload: Helper.payload_to_adts(buffer.payload, ctx.pads.input.caps)}
+
+        _other ->
+          buffer
+      end
```

Comment:
This is more `if else` than `case` as we have only two options

Reply:
There are more options, like LATM and I actually plan to include it, just not right now as we need this plugin to be released ASAP